libsolutil/JSON.cpp

```diff
@@ -43,116 +42,209 @@ namespace solidity::util
 namespace
 {
 
-/// StreamWriterBuilder that can be constructed with specific settings
-class StreamWriterBuilder: public Json::StreamWriterBuilder
+/// Takes a JSON value (@ _json) and removes all its members with value 'null' recursively.
+void removeNullMembersHelper(Json& _json)
 {
-public:
-	explicit StreamWriterBuilder(std::map<std::string, Json::Value> const& _settings)
+	if (_json.is_array())
 	{
-		for (auto const& iter: _settings)
-			this->settings_[iter.first] = iter.second;
+		for (auto& child: _json)
+			removeNullMembersHelper(child);
 	}
-};
-
-/// CharReaderBuilder with strict-mode settings
-class StrictModeCharReaderBuilder: public Json::CharReaderBuilder
-{
-public:
-	StrictModeCharReaderBuilder()
+	else if (_json.is_object())
 	{
-		Json::CharReaderBuilder::strictMode(&this->settings_);
+		for (auto it = _json.begin(); it != _json.end();)
+		{
+			if (it->is_null())
+				it = _json.erase(it);
+			else
+			{
+				removeNullMembersHelper(*it);
+				++it;
+			}
+		}
 	}
-};
+}
 
-/// Serialise the JSON object (@a _input) with specific builder (@a _builder)
-/// \param _input JSON input string
-/// \param _builder StreamWriterBuilder that is used to create new Json::StreamWriter
-/// \return serialized json object
-std::string print(Json::Value const& _input, Json::StreamWriterBuilder const& _builder)
+std::string trimRightAllLines(std::string const& input)
 {
-	std::stringstream stream;
-	std::unique_ptr<Json::StreamWriter> writer(_builder.newStreamWriter());
-	writer->write(_input, &stream);
-	return stream.str();
+	std::vector<std::string> lines;
+	std::string output;
+	boost::split(lines, input, boost::is_any_of("\n"));
+	for (auto& line: lines)
+	{
+		boost::trim_right(line);
+		if (!line.empty())
+			output += line + "\n";
+	}
+	return boost::trim_right_copy(output);
 }
 
-/// Parse a JSON string (@a _input) with specified builder (@ _builder) and writes resulting JSON object to (@a _json)
-/// \param _builder CharReaderBuilder that is used to create new Json::CharReaders
-/// \param _input JSON input string
-/// \param _json [out] resulting JSON object
-/// \param _errs [out] Formatted error messages
-/// \return \c true if the document was successfully parsed, \c false if an error occurred.
-bool parse(Json::CharReaderBuilder& _builder, std::string const& _input, Json::Value& _json, std::string* _errs)
+std::string formatLikeJsoncpp(std::string const& _dumped, JsonFormat const& _format)
```

Comment:
The idea for this was to just merge this PR, and then remove this in a subsequent PR accepting the test changes, right :-)? So I don't need to properly review it :-D?

Reply:
Exactly! Thats the plan!

Reply:
See #15079